src/QsCompiler/LlvmBindings/Extensions/MarshaledStringArray.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) .NET Foundation and Contributors. All Rights Reserved. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+using System;
+
+namespace LlvmBindings.Interop
+{
+    public unsafe struct MarshaledStringArray : IDisposable
+    {
+        public MarshaledStringArray(ReadOnlySpan<string> inputs)
+        {
+            if (inputs.IsEmpty)
+            {
+                this.Count = 0;
+                this.Values = null;
+            }
+            else
+            {
+                this.Count = inputs.Length;
+                this.Values = new MarshaledString[this.Count];
+
+                for (int i = 0; i < this.Count; i++)
+                {
+                    this.Values[i] = new MarshaledString(inputs[i].AsSpan());
+                }
+            }
+        }
+
+        public int Count { get; private set; }
```

Comment:
Count seems to always reflect the number of elements in Values. Do you need a separate field?

Reply:
Perhaps not. This code was copied almost verbatim from LLVMSharp (I only fixed warnings due to our strictly repo settings). The original is at https://github.com/dotnet/LLVMSharp/blob/ad498be4815dc4699981ba61c9ade30a740d5904/sources/LLVMSharp.Interop/Internals/MarshaledStringArray.cs. We could do more general cleanup across these files, but for now my focus has been on copying functionality rather than on doing any refactoring.